src/legacy/core_plugins/vis_type_timelion/public/helpers/arg_value_suggestions.ts

```diff
@@ -53,7 +56,7 @@ export function getArgValueSuggestions() {
     }
     const indexPatternTitle = get(indexPatternArg, 'value.text');
 
-    const { savedObjects } = await savedObjectsClient.find({
+    const { savedObjects } = await savedObjectsClient.find<SavedObjectIndexPattern>({
```

Comment:
I added an explicit dependency on the `SavedObjectIndexPattern` to track all places accessing the data plugin saved object. It should simplify further refactoring.